opm/input/eclipse/Schedule/Schedule.hpp

```diff
@@ -94,6 +94,8 @@ namespace Opm
         bool rptonly{false};
         bool gaslift_opt_active{false};
         std::optional<OilVaporizationProperties> oilVap;
+        double default_bhp_target_prod;
+        double default_bhp_limit_inj;
```

Comment:
In principle, although I've never seen it happen in a real case, I think the keyword **could** be entered multiple times in the SCHEDULE section, and with different targets/limits each time.  In order to support such usage, I think we need to make these into a [`ptr_member`](https://github.com/OPM/opm-common/blob/3470f41564d834e8a7e43f22c466aee033e08e02/opm/input/eclipse/Schedule/ScheduleState.hpp#L114) in [`ScheduleState`](https://github.com/OPM/opm-common/blob/3470f41564d834e8a7e43f22c466aee033e08e02/opm/input/eclipse/Schedule/ScheduleState.hpp#L361-L382) and use the `update()` mechanism in `Schedule::handleFBHPDEF()`.

Reply:
actually this is exactly what I am testing in the test - and it works.

Reply:
(i'm not insisting on my approach, just to point out that it actually works as-is).

Reply:
FBHPDEF is a dynamic SCHEDULE keywords. It might not exist in most of Deck files. It might be input in the later stage of the SCHEDULE and take effects afterwards. It can also be input multiple times to have different values at different report steps. 

I totally agree with @bska , it should be in the `ScheduleStatic`.

Reply:
It's not a dynamic schedule keyword in its normal interpretation.

It sets defaults which are USED by other dynamic schedule keywords upon their processing (if they have defaulted values) There is no sense having it in the state, the values are not used in any way after the states have been process. We'll just keep numbers in memory that are completely unused. 

I will add it to the state, no worries, but I still think you are wrong.

Reply:
> [`FBHPDEF`] sets defaults which are USED by other dynamic schedule keywords upon their processing (if they have defaulted values) 

At first I was persuaded by this argument and my original reservation about parallel contexts does not apply, but then I thought of `ACTIONX`.  Our implementation of this feature **currently** "replays" all keywords after an action triggers and that means we do need to hold all such dynamic keywords in `ScheduleState`.  Otherwise, we'll just use the last instance when processing earlier `WCON*` keywords in an action context.

Reply:
That is a valid point. Then we also have the "local case" although probably mostly theoretic. There is nothing stopping you from having multiple entries within a single report step. As we'll have to store the last value for the step there is still an issue in theory (but obviously very unlikely).

Reply:
done.